### PR TITLE
[Kernel] Expand MoE weight loading + Add Fused Marlin MoE Kernel

### PR DESCRIPTION
**Pull request by dsikka**
# Summary
- Expands weight loading to support grouped and per channel weight quantization. Cleans-up fp8 MoE to use the updated weight loading
- Adds Marlin Fused MoE Kernel for `w4a16` by @ElizaWszola 
- Adds `CompressedTensorsMoEMethod` to support MoE `w4a16` models from [llm-compressor](https://github.com/vllm-project/llm-compressor) and [compressed-tensors](https://github.com/neuralmagic/compressed-tensors)
- Tested using 2 and 4 TP with Mixtral

Next Steps:
- The `CompressedTensorsMoEMethod` is not leveraging the scheme structure in-place for compressed-tensors in order to keep the scope of this PR focused on the kernel + updated weight loading. Will be updated in a follow-up to use the scheme structure 

co-authored by @ElizaWszola, from Neural Magic

### CODE REVIEW

File: tests/quantization/test_compressed_tensors.py [hunk 1]
```diff
@@ -161,10 +161,3 @@ def test_compressed_tensors_kv_cache(vllm_runner):
     with vllm_runner(model_path, kv_cache_dtype="fp8") as llm:
         output = llm.generate_greedy("Hello world!", max_tokens=20)
         assert output
-
-
-def test_compressed_tensors_fused_moe(vllm_runner):
```
Comment (dsikka) [inline tests/quantization/test_compressed_tensors.py]: @ElizaWszola is there a reason we're disabling this test?

Comment (ElizaWszola) [reply tests/quantization/test_compressed_tensors.py]: I'll bring it back - accidentally deleted when cleaning up after merge

**Comment by dsikka** [comment]
> W4A16, the "4" is int4? or fp4?

Int4

**Review by mgoin — APPROVED** [review]

**Comment by Maximilianxu** [comment]
@dsikka @mgoin Hi, is there a UT for this kernel? I saw the API has differences from Triton-based `FusedMoE`.

**Comment by B-201** [comment]
@dsikka Hi! Thanks for your work. Do you have plans to support gptq models in the future?

**Comment by binxuan** [comment]
Hi, thanks for the work! I am wondering does this MoE kernel work on A100 GPU?

**Comment by mgoin** [comment]
@binxuan Yes it is supported on A100 (SM 8.0 and up)

**Comment by binxuan** [comment]
Thanks for confirming. I tried the mainline code, but got following error. I think the device name returned from get_device_name somehow was bytes instead of str.

> device_name = current_platform.get_device_name().replace(" ", "_")
> TypeError: a bytes-like object is required, not 'str'

 After fix this issue, got a second error from triton mentioning 
> AssertionError: fp8e4nv data type is not supported on CUDA arch < 89

**Comment by fengyang95** [comment]
If I want to use it with deepseek-v2, I saw that it uses fused_moe by default. Do I need to swap it out to get it running?

**Comment by dsikka** [comment]
> If I want to use it with deepseek-v2, I saw that it uses fused_moe by default. Do I need to swap it out to get it running?

Hi @fengyang95 - are you trying to run a W4A16 deepseek-v2 model?

**Comment by fengyang95** [comment]
> > If I want to use it with deepseek-v2, I saw that it uses fused_moe by default. Do I need to swap it out to get it running?
> 
> Hi @fengyang95 - are you trying to run a W4A16 deepseek-v2 model?

YES

**Comment by fengyang95** [comment]
Using H20 (sm_90) can start normally; is it also because it is currently not compatible with L40 (sm_89)? @dsikka

**Pull request merged**